proto/disaggregated.proto

```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
 message TryMarkDeleteResponse {
     S3LockResult result = 1;
 }
+
+/// For compute task dispatch and data exchange ///
+
+message DisaggregatedTaskMeta {
+    uint64 start_ts = 1; // start ts of a query
+    // query_ts + server_id + local_query_id to represent a global unique query.
+    uint64 query_ts = 2; // timestamp when start to execute query, used for TiFlash miniTSO schedule.
+    uint64 server_id = 3; // TiDB server id
+    uint64 local_query_id = 4; // unique local query_id if tidb don't restart.
+    int64 task_id = 5; // if task id is -1 , it indicates a tidb task.
+    string executor_id = 6; // the exectuor id
+}
+message DisaggReadError {
+	int32 code = 1;
+	string msg = 2;
+}
+
+message EstablishDisaggregatedTaskRequest {
+    DisaggregatedTaskMeta meta = 1;
+    string address = 2; // target address of this task.
+    int64 timeout = 3;
+    // The key ranges, Region meta that read node need to execute TableScan
+    repeated coprocessor.RegionInfo regions = 4;
+    int64 schema_ver = 5;
+    // Used for PartitionTableScan
+    repeated coprocessor.TableRegions table_regions = 6;
+    // The encoded TableScan + Selection. The write node can use it
+    // to build the rough set filter.
+    bytes encoded_plan = 7;
+}
+
+message EstablishDisaggregatedTaskResponse {
+    DisaggReadError error = 1;
+    // The read node needs to update its region cache about
+    // these regions
+    repeated metapb.Region retry_regions = 2;
+    // Write node maintains a snapshot with a lease time.
+    // Read node should read the delta pages
+    // (ColumnFileInMemory and ColumnFileTiny)
+    // along with this store_id and snapshot_id.
+    uint64 store_id = 3; // metapb.Store.id
```

Comment:
Where is snapshot_id?

Reply:
add in the below line, we use `DisaggTaskMeta` as snapshot_id now